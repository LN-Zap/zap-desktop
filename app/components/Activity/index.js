import Activity from './Activity'

export default Activity
