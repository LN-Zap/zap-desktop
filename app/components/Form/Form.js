import React from 'react'
import PropTypes from 'prop-types'

import Isvg from 'react-inlinesvg'
import x from 'icons/x.svg'

import Pay from './Pay'
import Request from './Request'

import styles from './Form.scss'

const FORM_TYPES = {
  PAY_FORM: Pay,
  REQUEST_FORM: Request
}

const Form = ({ formType, formProps, closeForm }) => {
  if (!formType) {
    return null
  }

  const FormComponent = FORM_TYPES[formType]
  return (
    <div className={styles.container}>
      <div className={styles.closeContainer}>
        <span onClick={closeForm}>
          <Isvg src={x} />
        </span>
      </div>
      <FormComponent {...formProps} />
    </div>
  )
}

Form.propTypes = {
  formType: PropTypes.string,
  formProps: PropTypes.object.isRequired,
  closeForm: PropTypes.func.isRequired
}

export default Form
