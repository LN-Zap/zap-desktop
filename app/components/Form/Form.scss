@import '../../variables.scss';

.outtercontainer {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 100%;
  height: 100vh;
  background: $white;
  z-index: 0;
  opacity: 0;
  transition: all 0.5s;

  &.open {
    opacity: 1;
    z-index: 20;
  }
}

.innercontainer {
  position: relative;
  height: 100vh;
  margin: 5%;
}

.esc {
  position: absolute;
  top: 0;
  right: 0;
  color: $darkestgrey;
  cursor: pointer;
  padding: 20px;
  border-radius: 50%;

  &:hover {
    color: $bluegrey;
    background: $darkgrey;
  }

  &:active {
    color: $white;
    background: $main;
  }

  svg {
    width: 32px;
    height: 32px;
  }
}

.content {
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  height: 75vh;
  justify-content: center;
  align-items: center;
}
