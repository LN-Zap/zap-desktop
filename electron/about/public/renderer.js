'use strict'
Object.defineProperty(exports, '__esModule', { value: true })

window.ipcRenderer.on('about-window:info', (_, info) => {
  const app_name = info.product_name
  const content = info.use_inner_html ? 'innerHTML' : 'innerText'
  document.title = `About ${app_name}`
  const title_elem = document.querySelector('.title')
  title_elem.innerText = `${app_name}`

  const copyright_elem = document.querySelector('.copyright')
  if (info.copyright) {
    copyright_elem[content] = info.copyright
  } else if (info.license) {
    copyright_elem[content] = `Distributed under ${info.license} license.`
  }
  const icon_elem = document.getElementById('app-icon')
  icon_elem.src = info.icon_path
  if (info.description) {
    const desc_elem = document.querySelector('.description')
    desc_elem[content] = info.description
  }

  if (info.css_path) {
    const css_paths = Array.isArray(info.css_path) ? info.css_path : [info.css_path]
    for (const css_path of css_paths) {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = css_path
      document.head.appendChild(link)
    }
  }

  if (info.use_version_info) {
    const versions = document.querySelector('.versions')
    const vs = process.versions
    for (const name of ['electron', 'chrome', 'node', 'v8']) {
      const tr = document.createElement('tr')
      const name_td = document.createElement('td')
      name_td.innerText = name
      tr.appendChild(name_td)
      const version_td = document.createElement('td')
      version_td.innerText = ' : ' + vs[name]
      tr.appendChild(version_td)
      versions.appendChild(tr)
    }
  }
})
//# sourceMappingURL=renderer.js.map
