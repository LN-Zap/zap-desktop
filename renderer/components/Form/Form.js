import React, { useRef } from 'react'
import PropTypes from 'prop-types'
import system from '@rebass/components'
import { styles } from 'styled-system'
import { Form as InformedForm } from 'informed'

// Create an html input element that accepts all style props from styled-system.
const FormInner = system(
  {
    extend: InformedForm,
  },
  ...Object.keys(styles)
)

const Form = ({ asyncValidators, onSubmit, onSubmitFailure, getApi, ...rest }) => {
  const formApi = useRef(null)

  const onBeforeSubmit = async values => {
    // pause submit if we have async validators set
    if (asyncValidators && asyncValidators.length) {
      await Promise.all(asyncValidators.map(v => v()))
    }

    const api = formApi.current
    if (api) {
      if (api.getState().invalid) {
        onSubmitFailure && onSubmitFailure()
      } else {
        onSubmit && onSubmit(values)
      }
    }
  }

  const getApiWrapper = api => {
    formApi.current = api
    getApi && getApi(api)
  }

  return (
    <FormInner
      {...rest}
      getApi={getApiWrapper}
      onSubmit={onBeforeSubmit}
      onSubmitFailure={onSubmitFailure}
    />
  )
}

Form.propTypes = {
  asyncValidators: PropTypes.array,
  getApi: PropTypes.func,
  onSubmit: PropTypes.func,
  onSubmitFailure: PropTypes.func,
}

export default Form
