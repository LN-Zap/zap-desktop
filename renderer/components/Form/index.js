export { BasicInput } from './Input'
export { BasicRange } from './Range'
export { BasicTextArea } from './TextArea'
export { BasicCheckbox } from './Checkbox'
export { BasicSelect } from './Select'
export Checkbox from './Checkbox'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export Input from './Input'
export InputLabel from './InputLabel'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export LndConnectionStringInput from './LndConnectionStringInput'
export LndConnectionStringEditor from './LndConnectionStringEditor'
export NodePubkeyInput from './NodePubkeyInput'
export IntegerInput from './IntegerInput'
export OpenDialogButton from './OpenDialogButton'
export OpenDialogInput from './OpenDialogInput'
export Radio from './Radio'
export RadioCard from './RadioCard'
export RadioGroup from './RadioGroup'
export Range from './Range'
export RowLabel from './RowLabel'
export Select from './Select'
export TextArea from './TextArea'
export Toggle from './Toggle'
export TransactionFeeInput, { TransactionSpeedDesc } from './TransactionFeeInput'
