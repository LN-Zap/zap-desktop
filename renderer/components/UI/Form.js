import system from '@rebass/components'
import { styles } from 'styled-system'
import { Form as InformedForm } from 'informed'

// Create an html input element that accepts all style props from styled-system.
const Form = system(
  {
    extend: InformedForm,
  },
  ...Object.keys(styles)
)

export default Form
