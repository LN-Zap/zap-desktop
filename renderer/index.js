import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-intl-redux'
import jstz from 'jstimezonedetect'
import translations from '@zap/i18n/translation'
import { getDefaultLocale } from '@zap/i18n'
import { configureStore, history } from './store/configureStore'
import Root from './containers/Root'

// Default the locale to English.
const defaultLocale = getDefaultLocale()

// Initialise the intl store with data from the users current locale.
const initialState = {
  intl: {
    locale: defaultLocale,
    messages: translations[defaultLocale],
    timeZone: jstz.determine().name(),
  },
}

// Set up the redux store.
const store = configureStore(initialState)
const MOUNT_NODE = document.getElementById('root')

const render = Component => {
  ReactDOM.render(
    <Provider store={store}>
      <Component history={history} />
    </Provider>,
    MOUNT_NODE
  )
}

render(Root)

if (module.hot) {
  module.hot.accept('./containers/Root', () => {
    render(Root)
  })
}
