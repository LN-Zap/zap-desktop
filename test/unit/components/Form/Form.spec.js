import React from 'react'
import renderer from 'react-test-renderer'
import { Form } from 'components/Form'

describe('component.UI.Form', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Form />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
