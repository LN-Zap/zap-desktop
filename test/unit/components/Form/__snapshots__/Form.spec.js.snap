// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Form should render correctly 1`] = `
<form
  className=""
  onKeyDown={[Function]}
  onReset={[Function]}
  onSubmit={[Function]}
/>
`;
